eispac/net/client.py

```diff
@@ -0,0 +1,135 @@
+    baseurl_fits = r'https://eis.nrl.navy.mil/level1/fits/%Y/%m/%d/eis_er_%Y%m%d_%H%M%S.fits'
+    pattern_fits = '{}/{year:4d}/{month:2d}/{day:2d}/eis_er_{:8d}_{hour:2d}{minute:2d}{second:2d}.{FileType}'
+    
+    @property
+    def info_url(self):
+        return 'https://eis.nrl.navy.mil/'
+
+    def search(self, *args, **kwargs):
+        metalist = []
+        matchdict = self._get_match_dict(*args, **kwargs)
+        all_filetypes = matchdict.get('FileType')
+        for ft in all_filetypes:
+            if 'h5' in ft:
+                baseurl = self.baseurl_hdf5
+                pattern = self.pattern_hdf5
+            else:
+                baseurl = self.baseurl_fits
+                pattern = self.pattern_fits
+
+            scraper = Scraper(baseurl, regex=True)
+            tr = TimeRange(matchdict['Start Time'], matchdict['End Time'])
+            filesmeta = scraper._extract_files_meta(tr, extractor=pattern, matcher={'FileType': ft})
+            filesmeta = sorted(filesmeta, key=lambda k: k['url'])
+            for i in filesmeta:
+                rowdict = self.post_search_hook(i, matchdict)
+                metalist.append(rowdict)
+
+        return QueryResponse(metalist, client=self)
+
+    def post_search_hook(self, i, matchdict):
+        # This makes the final display names of the file types nicer
+        filetype_mapping = {
+            'data.h5': 'HDF5 data',
+            'head.h5': 'HDF5 header',
+            'fits': 'FITS',
+        }
+        rd = super().post_search_hook(i, matchdict)
+        rd['FileType'] = filetype_mapping[rd['FileType']]
+        return rd
+    
+    @classmethod
+    def register_values(cls):
+        return {
+            a.Instrument: [('EIS', 'Extreme Ultraviolet Imaging Spectrometer')],
+            a.Physobs: [('intensity', 'Spectrally resolved intensity in detector units')],
+            a.Source: [('Hinode', 'The Hinode mission is a partnership between JAXA, NASA, and UKSA')],
+            a.Provider: [('NRL', 'U.S. Naval Research Laboratory')],
+            a.Level: [
+                ('1', 'EIS: The EIS client can only return level 1 data. Level 0 EIS data is available from the VSO.')
+            ],
+            FileType: [('data.h5', 'These files contain the actual intensity data in HDF5 format.'),
+                       ('head.h5', 'These files contain only the header metadata in HDF5 format.'),
+                       ('fits', 'These files contain both data and metadata in FITS format')],
+        }
+
+    @classmethod
+    def _attrs_module(cls):
+        # Register EIS specific attributes with Fido
+        return 'eispac', 'eispac.net.attrs'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+        """
+        Check if this client can handle a given Fido query.
+        Returns
+        -------
+        bool
+            True if this client can handle the given query.
+        """
+        required = {a.Time, a.Instrument, a.Source}
```

Comment:
Do you need to make Source required?

Reply:
There could be another future instrument called EIS? 🤷

Reply:
That doesn't mean you *need* to support source, if someone didn't provide source then they would get both. They could optionally use source to disambiguate.